falcon/asgi/_asgi_helpers.py

```diff
@@ -14,6 +14,51 @@
 
 import inspect
 
+from falcon.errors import UnsupportedError, UnsupportedScopeError
+from falcon.util.misc import _lru_cache_safe
+
+
+@_lru_cache_safe(maxsize=16)
```

Comment:
I don't remember what happens with the lru cache if the function emits an error. is the error cached or the function is called again each time?

Reply:
Errors are not cached:
```python
>>> import time
>>> import functools
>>> @functools.lru_cache
... def test(n):
...     if n == 42:
...         raise RuntimeError(f'[{time.time()}] {n} is unsupported')
...     return n
... 
>>> test(1)
1
>>> test(2)
2
>>> test(42)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in test
RuntimeError: [1614715125.4782972] 42 is unsupported
>>> test(42)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in test
RuntimeError: [1614715126.8562453] 42 is unsupported
```

OTOH both behaviours are acceptable for us I guess?